activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -303,6 +303,13 @@ def +(other)
         result = utc + other
         result.in_time_zone(time_zone)
       end
+    rescue TypeError
+      ActiveSupport.deprecator.warn(
+        "Adding an instance of #{other.class} to an instance of #{utc.class} is deprecated. This behavior will raise " \
+        "a `TypeError` in Rails 8.0."
+      )
+      result = utc.acts_like?(:date) ? utc.since(other) : utc + other rescue utc.since(other)
```

Comment:
As far as I know `utc.acts_like?(:date)` should always be false, and we've already tried utc + other and know it raises a TypeError.

Can this be ...?

```suggestion
      result = utc.since(other)
```